### PR TITLE
[doc] Use "atan2(y, x)" instead of "atan2(x, y)" (#1816)

### PR DESCRIPTION
**Pull request by new-TonyWang**
Related issue = #1816 
Modified  one "atan2(x, y)" in "docs/arithmetics.rst" and four "atan2(x, y)" in "docs/write_test.rst"

### CODE REVIEW

File: docs/write_test.rst [hunk 1]
```diff
@@ -166,7 +166,7 @@ Use a comma-separated list for multiple input values:
 
     @pytest.mark.parametrize('x,y', [(1, 2), (1, 3), (2, 1)])
```
Comment (archibate) [inline docs/write_test.rst:167]:
```suggestion
    @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
```
nit :)

File: docs/write_test.rst [hunk 2]
```diff
@@ -191,7 +191,7 @@ Use two separate ``parametrize`` to test **all combinations** of input arguments
     @pytest.mark.parametrize('y', [1, 2])
     # same as:  .parametrize('x,y', [(1, 1), (1, 2), (2, 1), (2, 2)])
```
Comment (archibate) [inline docs/write_test.rst:192]:
```suggestion
    # same as:  .parametrize('y,x', [(1, 1), (1, 2), (2, 1), (2, 2)])
```
for consistance.

File: docs/write_test.rst (hunk 1 above)
Comment (new-TonyWang) [reply docs/write_test.rst:167]: solve!

File: docs/write_test.rst (hunk 2 above)
Comment (new-TonyWang) [reply docs/write_test.rst:192]: U are right!

File: docs/write_test.rst [hunk 3]
```diff
@@ -202,7 +202,7 @@ Use two separate ``parametrize`` to test **all combinations** of input arguments
         r[None] = x
         s[None] = y
```
[the remark this answers is not in the document]
Comment (new-TonyWang) [reply docs/write_test.rst:203]: Haha, one issue, two bugs got fixed

**Review by new-TonyWang — COMMENTED** [review]
My first time to fork a work, excited!

**Review by new-TonyWang — COMMENTED** [review]
OK

**Review by archibate — APPROVED** [review]
Cool!

**Review by yuanming-hu — APPROVED** [review]
Awesome! LGTM. Thank you so much!

Welcome to the Taichi developer community! :-)

File: docs/write_test.rst [hunk 4]
```diff
@@ -164,20 +164,20 @@ Use a comma-separated list for multiple input values:
...
         r = ti.field(ti.f32, ())
         s = ti.field(ti.f32, ())
 
         @ti.kernel
         def foo():
-            r[None] = ti.atan2(r[None])
+            r[None] = ti.atan2(r[None], s[None])
 
         r[None] = x
         s[None] = y
```
Comment (xumingkuan) [inline docs/write_test.rst:178]: I wonder if the `x` and `y` here need to be swapped too?

Comment (new-TonyWang) [reply docs/write_test.rst:178]: Yes, u are right. I think I should double check this file.

Comment (new-TonyWang) [reply docs/write_test.rst:178]: Actually this script  cannot pass the test, due to line 5. 
```python
import taichi as ti
import pytest
import math
@pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])#fixture 'taichi_archs' not found
@ti.test()
def test_atan2(y, x):
    r = ti.field(ti.f32, ())
    s = ti.field(ti.f32, ())
    print(x)
    print(y)
    @ti.kernel
    def foo():
        r[None] = ti.atan2(r[None], s[None])
    r[None] = y
    s[None] = x
    foo()
    assert r[None] == math.atan2(y, x)
```
Then I modify it to the  second script and the terminal show that r[None] and math.atan2(y, x) are not the same.

```python
import taichi as ti
    import pytest
    import math

    @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
    @ti.test()
    def test_atan2(y, x):
        r = ti.field(ti.f32, ())
        s = ti.field(ti.f32, ())

        @ti.kernel
        def foo():
            r[None] = ti.atan2(r[None], s[None])

        r[None] = y
        s[None] = x
        foo()
        assert r[None] == math.atan2(y, x)#not the same!!!
```
I modify it to the  third script and finally this script passes the test.
```python
import taichi as ti
import pytest
import math
@ti.all_archs
@pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
def test_atan2(y, x):
    r = ti.field(ti.f32, ())
    s = ti.field(ti.f32, ())
    print(x)
    print(y)
    @ti.kernel
    def foo():
        r[None] = ti.atan2(r[None], s[None])
    r[None] = y
    s[None] = x
    foo()
    assert ti.approx(r[None]) == math.atan2(y, x)
```

Comment (xumingkuan) [reply docs/write_test.rst:178]: > Then I modify it to the second script and the terminal show that r[None] and math.atan2(y, x) are not the same.

Interesting! I also found that they're not the same:
```
FAILED test2.py::test_atan2[2-1] - assert 1.1071487665176392 == 1.1071487177940904
```
But I can't run the script either. I modified `@ti.test()` in the second script to `@ti.all_archs` to run it.

@archibate Do you have any idea on `@ti.test()` itself runs successfully but `@pytest.mark.parametrize` together with `@ti.test` fails?

**Comment by xumingkuan** [comment]
Test case:
```python
import taichi as ti
import pytest
import math

@pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
@ti.test()
def test_atan2(y, x):
    r = ti.field(ti.f32, ())
    s = ti.field(ti.f32, ())

    @ti.kernel
    def foo():
        r[None] = ti.atan2(r[None], s[None])

    r[None] = y
    s[None] = x
    foo()
    assert r[None] == math.atan2(y, x)#not the same!!!
```
Log:
```
(base) C:\Users\xmk\Desktop\python_test>pytest -s test2.py
========================================================================================================== test session starts ===========================================================================================================
platform win32 -- Python 3.7.4, pytest-6.0.1, py-1.9.0, pluggy-0.12.0
rootdir: C:\Users\xmk\Desktop\python_test
plugins: rerunfailures-9.0, xdist-1.32.0, arraydiff-0.3, cov-2.10.0, doctestplus-0.3.0, forked-1.1.3, openfiles-0.3.2, remotedata-0.3.2
collecting ... [Taichi] mode=development
[Taichi] <dev mode>, llvm 10.0.0, commit e0ac1d86, win, python 3.7.4
[W 09/30/20 00:16:30.115] [shell.py:<module>@7] `sourceinspect` not installed!
[W 09/30/20 00:16:30.116] [shell.py:<module>@9] Without this package Taichi may not function well in Python IDLE interactive shell, Blender scripting module and Python native shell.
[W 09/30/20 00:16:30.116] [shell.py:<module>@11] Please run `python3 -m pip install sourceinspect` to install.
[I 09/30/20 00:16:30.117] [shell.py:_shell_pop_print@31] Graphical python shell detected, using wrapped sys.stdout
collected 3 items                                                                                                                                                                                                                         

test2.py EEE

================================================================================================================= ERRORS =================================================================================================================
___________________________________________________________________________________________________ ERROR at setup of test_atan2[1-2] ____________________________________________________________________________________________________
file C:\Users\xmk\Desktop\python_test\test2.py, line 5
  @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
  @ti.test()
  def test_atan2(y, x):
E       fixture 'taichi_archs' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, t
mp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

C:\Users\xmk\Desktop\python_test\test2.py:5
___________________________________________________________________________________________________ ERROR at setup of test_atan2[1-3] ____________________________________________________________________________________________________
file C:\Users\xmk\Desktop\python_test\test2.py, line 5
  @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
  @ti.test()
  def test_atan2(y, x):
E       fixture 'taichi_archs' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, t
mp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

C:\Users\xmk\Desktop\python_test\test2.py:5
___________________________________________________________________________________________________ ERROR at setup of test_atan2[2-1] ____________________________________________________________________________________________________
file C:\Users\xmk\Desktop\python_test\test2.py, line 5
  @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
  @ti.test()
  def test_atan2(y, x):
E       fixture 'taichi_archs' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, t
mp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

C:\Users\xmk\Desktop\python_test\test2.py:5
============================================================================================================ warnings summary ============================================================================================================
C:\Users\xmk\Desktop\taichi\python\taichi\testing.py:109
  C:\Users\xmk\Desktop\taichi\python\taichi\testing.py:109: PytestUnknownMarkWarning: Unknown pytest.mark.taichi - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stabl
e/mark.html
    @pytest.mark.taichi(*args, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================================================================== short test summary info =========================================================================================================
ERROR test2.py::test_atan2[1-2]
ERROR test2.py::test_atan2[1-3]
ERROR test2.py::test_atan2[2-1]
====================================================================================================== 1 warning, 3 errors in 1.17s ======================================================================================================

(base) C:\Users\xmk\Desktop\python_test>
```

**Comment by xumingkuan** [comment]
I found that if the test file is in `taichi/tests/python`, there won't be the `fixture 'taichi_archs' not found` issue...

However, `ti.approx(r[None])` is still necessary to let the test pass.

**Comment by new-TonyWang** [comment]
Yes, the ```ti test [filename]``` is functional only the test files are located in ```taichi\tests\python```  (This has already mentioned in the doc), but any body who doesn't cone this repository will not able to test their taichi programs. I think I should first modify  this
``` python
ti.approx(r[None]) == math.atan2(y, x)
```
and further find a way to solve this test case path issue

**Pull request merged**